### ISSUE TITLE
readQuery does not include changes made by writeFragment

### ISSUE THREAD

**Issue by yattias** [apollographql/apollo-client#1598]
I think I am misunderstanding an important point in regards to local cache function [documented here](http://dev.apollodata.com/core/read-and-write.html)

**Intended outcome:**
To fully update one document in my cache in such a way where if the document is updated by ID with `writeFragment`, executing a subsequent `readQuery` should include my updated document.

The use case here is that my app is making local changes to documents via `writeFragment`. I need to then fetch a list of all of my locally changed documents - Ideally via something like `readQuery`.

**Actual outcome:**
`writeFragment` is executed successfully. Calling `readQuery` then returns the original result of the query when the app was initialized before local changes were made rather than including my recent changes made by `writeFragment`.

**How to reproduce the issue:**
```js
graphqlClient.writeFragment({
        id: id,
        fragment: gql`
          fragment attributes on Asset {
            plainText
            _hasUnsyncedChanged
          }
        `,
        data: data
      });
```

Then when I run the query that originally populated the cache, I don't see any of my changes reflected:

```js
    data = graphqlClient.readQuery({
      query:  gql`
        query getAssets {
          assets {
            plainText
            _hasUnsyncedChanged
            id
          }
        }`
    });
```

Thank you!

**Comment by helfer** [comment]
Thanks @yattias! Could you put this into a repro using the [error template](https://github.com/apollographql/react-apollo-error-template)?

**Comment by helfer** [comment]
How are you making sure that the Asset you're inserting in the first query becomes part of the assets in the second query? Apollo Client can't magically know that `assets` is a list of all Assets, you have to manually insert the asset into the list of Assets.

**Comment by yattias** [comment]
Hey @helfer sorry, there was mutation in there i forgot to add.

You can close this actually, the issue has been resolved once I updated to version 1.0.2. This is no longer an issue for me 👍
